plugins/inputs/logfile/tail/tail_queue_test.go

```diff
@@ -0,0 +1,27 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package tail
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestTailerEnqueueAndDequeue(t *testing.T) {
+	tailerEnqueue := NewTailerFifoQueue()
+	tailerEnqueue.Enqueue()
+	tailerEnqueue.Enqueue()
+	assert.Equal(t, 2, tailerEnqueue.Size())
+
+	ok := tailerEnqueue.Dequeue()
+	assert.Equal(t, 1, tailerEnqueue.Size())
+	assert.Equal(t, true, ok)
```

Comment:
If you don't intend to change the interface for your counter, please simplify this to 
```suggestion
	assert.True(t, ok)
```

Reply:
Sorry for this one too. I'm still in the middle of transitioning and have not planned to touch this in the morning.